### PR TITLE
added 'tree' command

### PR DESCRIPTION
**Pull request by y0014984**
Merging this Pull Request will add the 'tree' command to armaOS.

### CODE REVIEW

File: addons/armaos/functions/fnc_os_tree.sqf [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    try
+    {
+        // function does not return a bool but throws an exception if permissions insufficient
+        [_filesystem, _user, _permissionNeeded] call AE3_filesystem_fnc_hasPermission;
+
+        {
+            // Hashmap forEach variables: KEY = _x && VALUE = _y
+
+            private _treeString = "";
+            for "_i" from 0 to _level do
```
Comment (GermanHydrogen) [inline addons/armaos/functions/fnc_os_tree.sqf:59]: Could be simplified to

```sqf

for "_i" from 0 to _level - 1 do
{
    _treeString = _treeString + "|  ";
};
_treeString = _treeString + "|--";
```

File: addons/armaos/functions/fnc_os_tree.sqf [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+private _filesystem = _computer getVariable "AE3_filesystem";
+
+private _current = [_pointer, _filesystem] call AE3_filesystem_fnc_resolvePntr;
+
+private _terminal = _computer getVariable "AE3_terminal";
+private _user = _terminal get "AE3_terminalLoginUser";
+
+private _level = 0;
+
+private _fnc_tree_recursive =
```
Comment (GermanHydrogen) [inline addons/armaos/functions/fnc_os_tree.sqf:38]: Subfunction should be defined at the top of the function for consistency

File: addons/armaos/functions/fnc_os_tree.sqf [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+};
+
+private _result = [_pointer, _current, _user, _level] call _fnc_tree_recursive;
+
+private _totalResults = _result select 0;
+private _missingPermissions = _result select 1;
+
+if (_missingPermissions > 0) then
+{
+	_totalResults append [format [localize "STR_AE3_ArmaOS_Exception_CantScanFolderMissionPermissions", _missingPermissions]];
```
Comment (GermanHydrogen) [inline addons/armaos/functions/fnc_os_tree.sqf:98]: Alternatively, we could simply hide the contents of dirs for which the user does not have permission. I don't know what would be more consistent.

**Review by GermanHydrogen — CHANGES_REQUESTED** [review]
Looks good in general, only small comments

File: addons/armaos/functions/fnc_os_tree.sqf (hunk 1 above)
Comment (y0014984) [reply addons/armaos/functions/fnc_os_tree.sqf:59]: Looks good. Will implement that.

File: addons/armaos/functions/fnc_os_tree.sqf (hunk 2 above)
Comment (y0014984) [reply addons/armaos/functions/fnc_os_tree.sqf:38]: Ok. Will do that.

File: addons/armaos/functions/fnc_os_tree.sqf (hunk 3 above)
Comment (y0014984) [reply addons/armaos/functions/fnc_os_tree.sqf:98]: I copied the behavior from the `find` command. There we do the same. So I think this is consistent.

**Review by GermanHydrogen — APPROVED** [review]
LGTM

**Pull request merged**